pkgs/development/python-modules/web3/default.nix

```diff
@@ -33,24 +33,30 @@ buildPythonPackage rec {
     hash = "sha256-RNWCZQjcse415SSNkHhMWckDcBJGFZnjisckF7gbYY8=";
   };
 
+  # Note: to reflect the extra_requires in main/setup.py.
+  passthru.optional-dependencies = {
+    # tester = [ eth-tester py-geth ];
+    # linter = [ ... ];
+    # docs = [ ... ];
+    # dev = [ ... ];
+    ipfs = [ ipfshttpclient ];
+  };
+
   propagatedBuildInputs = [
     aiohttp
     eth-abi
     eth-account
-    eth-hash
-    eth-rlp
+    eth-hash ] ++ eth-hash.optional-dependencies.pycryptodome ++ [
     eth-typing
     eth-utils
     hexbytes
-    ipfshttpclient
     jsonschema
     lru-dict
     protobuf
     requests
-    websockets
-  ] ++ lib.optionals (pythonOlder "3.8") [
```

Comment:
This change should be reverted because there is a conditional branch as of v6.5.0.

Reply:
And we don't need `typing-extension` in python 3.8 and above.

Reply:
Okay, indeed.

Thanks for the helping so far. Please have a look again!